daliuge-engine/dlg/manager/rest.py

```diff
@@ -348,6 +373,44 @@ def getNodeSessions(self, node):
         with NodeManagerClient(host=node) as dm:
             return dm.sessions()
 
+    def _tarfile_write(self, tar, headers, stream):
+        file_header = headers.getheader('Content-Disposition')
+        length = headers.getheader('Content-Length')
+        _, params = cgi.parse_header(file_header)
+        filename = params['filename']
+        info = tarfile.TarInfo(filename)
+        info.size = int(length)
+
+        content = []
+        while True:
+            buffer = stream.read()
```

Comment:
Because its a tcp stream there is no guarantee that a single read will get all the data unless there is code underneath this library that is accumulating?

Reply:
I think ill leave it for now and we can review again.